File: ClientApp/ClientApp/Models/Responses/CreateRoomResponse.cs
using ClientApp.Models.ResponseErrors;
using Newtonsoft.Json;

namespace ClientApp.Models.Responses
{
    class CreateRoomResponse : Response<CreateRoomResponseErrors>
    {
        private CreateRoomResponse() { }
        [JsonProperty]
        public RoomDataModel RoomData { get; private set; }
    }
}

File: M_Trivia/M_Trivia/CreateRoomResponse.cpp
#include "CreateRoomResponse.hpp"

#include "json.hpp"

CreateRoomResponse::CreateRoomResponse(std::unique_ptr<CreateRoomResponseErrors> errors, const RoomData& roomData)
    : Response(std::move(errors)), roomData(roomData) {
}


nlohmann::json CreateRoomResponse::getJson() const {
	nlohmann::json j = Response::getJson();
	j["RoomData"] = roomData;
	return j;
}



